kmm/kmm-post/src/commonMain/kotlin/com/gchristov/newsfeed/kmmpost/PostModule.kt

```diff
@@ -3,26 +3,37 @@ package com.gchristov.newsfeed.kmmpost
 import com.gchristov.newsfeed.kmmcommondi.DiModule
 import com.gchristov.newsfeed.kmmcommondi.inject
 import com.gchristov.newsfeed.kmmpostdata.PostDataModule
+import com.gchristov.newsfeed.kmmpostdata.PostRepository
+import com.gchristov.newsfeed.kmmpostdata.usecase.DecoratePostUseCase
 import kotlinx.coroutines.Dispatchers
 import org.kodein.di.DI
 import org.kodein.di.bindFactory
+import org.kodein.di.bindProvider
 import org.kodein.di.instance
 
 object PostModule : DiModule() {
     override fun name() = "kmm-post"
 
     override fun bindLocalDependencies(builder: DI.Builder) {
         builder.apply {
+            //TODO: provider or singleton?
```

Comment:
@gchristov I read a bit on KodeIn and this may fit more the `singleton` case, as we only want one and not every time is called? But the examples in the code have all providers so I left it as provider

Reply:
I think it's okay to have the use case be created every time as it's relatively cheap - there's almost no initialization cost. Also in some cases it might store local state which we wouldn't want to share with a singleton.